src/main/kotlin/day11/FileOperations.kt

```
@@ -0,0 +1,43 @@
+    )
+    fun copy(fromPath: String, toPath: String)
+}
+
+class DefaultFileOperations : FileOperations {
+    override fun copy(fromPath: String, toPath: String) {
+        val paths = pathResolver(fromPath, toPath)
+        val from = paths.first.toFile()
+        val to = paths.second.toFile()
+
+        if (!to.canWrite()) {
+            throw IllegalArgumentException()
```

detekt (1.23.7)

[warning] A call to the default constructor of an exception was detected. Instead one of the constructor overloads should be called. This allows to provide more meaningful exceptions.

(detekt.exceptions.ThrowingExceptionsWithoutMessageOrCause)